### PR TITLE
upgrade to Julia 1.x

### PR DESCRIPTION
**Pull request by johnnychen94**
The update work is split into several PRs(e.g., #27, #31) and will be merged into this PR/branch first. There will be a lot of changes, so it's not suitable to directly review this.

Basically, the roads are:

- [x] fix 0.7 deprecations (#27)
- [x] fix 1.x deprecations (#31)
- [x] bug: #38 
- [x] bug: #39
- [ ] upgrade to OffsetArray v1.0
- [ ] update docs

When all things get settled done, I will merge this branch into master.

### CODE REVIEW

**Comment by johnnychen94** [comment]
Commits 40820af and 03d97a0 were pushed to master and rebased to this branch; then I realized that they won't actually work before this PR is ready. To disable daily notification of failed GitHub action jobs, I deleted these two commits from the master branch so that there will be no scheduled jobs running.

To keep track of history, this PR shouldn't be merged with "squash and merge" option.

File: src/operations/distortion.jl [hunk 1]
```diff
@@ -93,7 +93,7 @@ augment(img, ElasticDistortion(15, 15; scale = 0.1))
 # distort everything more smoothly.
 augment(img, ElasticDistortion(10, 10; sigma = 4, iter=3, border=true))
 ```
-""" ->
+"""
```
Comment (Evizero) [inline src/operations/distortion.jl:96]: are you sure?

File: test/operations/tst_mapfun.jl [hunk 2]
```diff
@@ -101,13 +103,14 @@ end
...
-                @test typeof(indices(img_in)) <: NTuple{2,Base.OneTo} ? typeof(res) <: Array : typeof(res) <: OffsetArray
+                @test typeof(axes(img_in)) <: NTuple{2,Base.OneTo} ? typeof(res) <: Array : typeof(res) <: OffsetArray
                 res = @inferred(Augmentor.applyeager(AggregateThenMapFun(mean, (x,a)->x-a), img_in))
                 @test res == img_out
                 @test typeof(res) == typeof(img_out)
             end
             img = OffsetArray(rgb_rect, -2, -1)
-            res = @inferred(Augmentor.applyeager(AggregateThenMapFun(mean, (x,a)->x-a), img))
+            @test_broken false # FIXME: type stability below from "mean"
+            res = (Augmentor.applyeager(AggregateThenMapFun(mean, (x,a)->x-a), img))
```
Comment (Evizero) [inline test/operations/tst_mapfun.jl]: Out of interest: Any idea where this issue came from? `mean` itself?

**Review by Evizero — APPROVED** [review]

File: src/operations/distortion.jl (hunk 1 above)
Comment (johnnychen94) [reply src/operations/distortion.jl:96]: Ah, these two changes are made in 8dbe2b1 (#27 ) and my changes #31 is built on top of that.

I haven't observe such `->` usage after Julia 1.0; so I guess it's an deprecated syntax

```julia
help?> ElasticDistortion
```
works normally.

File: test/operations/tst_mapfun.jl (hunk 2 above)
Comment (johnnychen94) [reply test/operations/tst_mapfun.jl]: This test isn't broken anymore.

**Comment by johnnychen94** [comment]
Since this branch is functioning now with all deprecations fixed, I'm merging this branch to master. From now on, compat upgrades(e.g., OffsetArrays 1.0 is a breaking change) and new features can be added to the master branch separately.

**Comment by johnnychen94** [comment]
@Evizero since most CI are moved to github action, can you uninstall appveyor?

Submitting coverage report via github action doesn't work for PRs from other forks. (It's an upstream github action issue), hence travis is still required to upload coverage report

**Comment by Evizero** [comment]
done

**Pull request merged**